src/app/shared/spaces.service.spec.ts

```diff
@@ -0,0 +1,166 @@
+import { TestBed } from '@angular/core/testing';
+import { LocalStorageService } from 'angular-2-local-storage';
+import { Broadcaster } from 'ngx-base';
+import { Context, Contexts, Space, SpaceService } from 'ngx-fabric8-wit';
+import { User } from 'ngx-login-client';
+import { ConnectableObservable, Observable } from 'rxjs';
+import { createMock } from 'testing/mock';
+import { ExtProfile, ProfileService } from '../profile/profile.service';
+import { SpacesService } from './spaces.service';
+
+describe('SpacesService', () => {
+
+  let mockSpace: Space;
```

Comment:
My preference is to use `TestBed.get` to access these mock instances from within individual specs, rather than holding onto top-level references. This goes nicely with the `useFactory` trick you're already using below for the `providers`.

Actually I see you do use `TestBed.get` in the specs below some of the time, so I'd prefer to use it consistently and avoid these top-level references.

Reply:
@aptmac is this something you can address or explain why not?

Reply:
Ah I should have commented when I fixed it, I already made all the changes to better used `TestBed.get()` in a follow up commit, see below: https://github.com/fabric8-ui/fabric8-ui/pull/3201/commits/2528d95a6b8f3d9169ab7dba78c77c667b4b22ee

Reply:
Yea I reviewed that commit and approved the change. The remaining top-level references are just for mock data, not service references, which I think is fine.

Reply:
thanks